### PR TITLE
Backend: Split out command executor interface.

### PR DESCRIPTION
**Pull request by mook-as**
This adds an interface for executing commands within the VM.  This PR also includes switching K3sHelper to use the new interface as a sample.

This is part of #1997.

### CODE REVIEW

File: src/backend/k3sHelper.ts [hunk 1]
```diff
@@ -940,7 +940,7 @@ export default class K3sHelper extends events.EventEmitter {
    * deleted.
    * @param execAsRoot A function to run commands on the VM as root.
```
Comment (ericpromislow) [inline src/backend/k3sHelper.ts:941]: This @param is now obsolete

File: src/backend/lima.ts [hunk 2]
```diff
@@ -416,7 +416,7 @@ export default class LimaBackend extends events.EventEmitter implements K8s.Kube
   get ipAddress(): Promise<string | undefined> {
     return (async() => {
       // Get the routing map structure
-      const state = await this.limaWithCapture('shell', '--workdir=.', MACHINE_NAME, 'cat', '/proc/net/fib_trie');
+      const state = await this.execCommand({ capture: true }, 'cat', '/proc/net/fib_trie');
```
Comment (ericpromislow) [inline src/backend/lima.ts:419]: I'm a bit confused why you replace `limaWithCapture` with `execCommand` in a few places, but left `limaWithCapture` in others (although it makes sense to have calls to `limaWithCapture` inside `execCommand`.

File: src/backend/backend.ts [hunk 3]
```diff
@@ -178,3 +181,40 @@ export interface VMBackend {
    */
   noModalDialogs: boolean;
 }
+
+/**
+ * execOptions is options for VMExecutor.
+ */
+export type execOptions = childProcess.CommonOptions & {
+  /** Output encoding; defaults to utf16le. */
```
Comment (ericpromislow) [inline src/backend/backend.ts:189]: This comment is true for WSL only, right? It's not set for lima, so does that mean it defaults to utf-8 ?

**Review by ericpromislow — CHANGES_REQUESTED** [review]
Just a couple of small changes, and a request for a clarification

File: src/backend/backend.ts (hunk 3 above)
Comment (mook-as) [reply src/backend/backend.ts]: Fixed in 0d89d352fc0d1745ada9606ca00967f364d59e5d.

File: src/backend/k3sHelper.ts (hunk 1 above)
Comment (mook-as) [reply src/backend/k3sHelper.ts]: Fixed in 7994a25a8e04a7f0435170bfabd408e23ffbc116.

File: src/backend/lima.ts (hunk 2 above)
Comment (ericpromislow) [reply src/backend/lima.ts:419]: thx

**Review by ericpromislow — CHANGES_REQUESTED** [review]
Please fix the lint errors:
```

/Users/ericp/workspace/rancher/desktop/src/backend/lima.ts
    21:24  warning  'BackendError' is defined but never used. Allowed unused vars must match /^_./u  @typescript-eslint/no-unused-vars
  1587:9   warning  'commandArgs' is defined but never used. Allowed unused vars must match /^_./u   @typescript-eslint/no-unused-vars

✖ 2 problems (0 errors, 2 warnings)
```

Everything else looks fine

**Review by ericpromislow — CHANGES_REQUESTED** [review]
Needs rebasing to pull in recent changes in the UI

**Pull request merged**